custom_components/google_home/sensor.py

```diff
@@ -68,42 +70,28 @@ async def async_setup_entry(
     return True
 
 
-class GoogleHomeSensorMixin:
-    """Adds basic functions to sensors"""
-
-    def get_device(self):
-        """Return the device matched by device name
-        from the list of google devices in coordinator_data"""
-        return next(
-            (
-                device
-                for device in self.coordinator.data
-                if device.name == self.device_name
-            ),
-            None,
-        )
-
-    @staticmethod
-    def as_dict(obj_list: List[Any]) -> List[Dict[Any, Any]]:
-        """Return list of objects represented as dictionaries """
-        return [obj.__dict__ for obj in obj_list]
+class GoogleHomeDeviceSensor(GoogleHomeBaseEntity):
+    """Google Home Device sensor."""
 
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return f"{self.device_name} {LABEL_DEVICE}"
 
-class GoogleHomeDeviceSensor(GoogleHomeSensorMixin, GoogleHomeDeviceEntity):
-    """GoogleHome Device Sensor class."""
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID to use for this entity."""
+        return f"{self.device_name}/{LABEL_DEVICE}"
 
-    def __init__(
-        self, coordinator: DataUpdateCoordinator, device_name: str, auth_token: str
-    ):
-        """Initialize the sensor."""
-        super().__init__(coordinator, device_name)
-        self.auth_token = auth_token
+    @property
+    def icon(self) -> str:
+        """Return the icon of the sensor."""
+        return ICON_TOKEN
```

Comment:
How about we set on a class level:
```python
class GoogleHomeDeviceSensor(GoogleHomeBaseEntity):
  LABEL = LABEL_DEVICE
  ICON = ICON_TOKEN
  ....
```
and then move all those reperative `def name; def unique_id; def icon` back to `GoogleHomeBaseEntity` where these methods would look like:
```python
    @property
    def name(self) -> str:
        """Return the name of the sensor."""
        return f"{self.device_name} {self.LABEL}"
....
```
It would help us avoid repetitions and keep cleaner :)

Reply:
Parent class can't really use constants defined in subclasses. But we can use abstract property here.